src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs

```diff
@@ -59,22 +61,25 @@ internal static partial class SettingsUpdateHelper
             var updatedText = originalText;
             var settings = settingsToUpdate.Select(x => TryGetOptionValueAndLanguage(x.option, x.value, optionSet))
                                            .Where(x => x.success)
-                                           .Select(x => (x.option, x.value, x.language))
+                                           .Select(x => (x.option, x.value, x.ids, x.severity, x.language))
                                            .ToList();
 
             return TryUpdateAnalyzerConfigDocument(originalText, filePath, settings);
 
-            static (bool success, string option, string value, Language language) TryGetOptionValueAndLanguage(IOption2 option, object value, OptionSet optionSet)
+            static (bool success, string option, string value, ImmutableArray<string> ids, string? severity, Language language) TryGetOptionValueAndLanguage(IOption2 option, object value, OptionSet optionSet)
             {
                 if (option.StorageLocations.FirstOrDefault(x => x is IEditorConfigStorageLocation2) is not IEditorConfigStorageLocation2 storageLocation)
                 {
-                    return (false, null!, null!, default);
+                    return (false, null!, null!, ImmutableArray<string>.Empty, null, default);
                 }
 
                 var optionName = storageLocation.KeyName;
                 var optionValue = storageLocation.GetEditorConfigStringValue(value, optionSet);
+                var language = option.IsPerLanguage ? Language.CSharp | Language.VisualBasic : Language.CSharp;
```

Comment:
i don't really get thsi.  if it's not per langauge, then i would expect it to be C#/VB (As that's waht a normal option affects).  if it is per-language, then i woudl expect it to pick the language based on what language the otion is intended for.

Reply:
@CyrusNajmabadi This is tracked by https://github.com/dotnet/roslyn/issues/52696.

Reply:
This is the reverse of the normal case our options operate under. Normally you get an option and you want to know what it applies to in the context of a project. For editorconfig we have the reverse problem. How should a per-language option be persisted? Do users want their single file to have duplicate language for each language type in their project? Or do they want a unified header? 

I've chosen to, where possible, write out options that can apply to both languages together.